draft-wang-ppm-differential-privacy.md

```diff
@@ -400,37 +399,270 @@ with the correct noise type.
+  response mechanisms have a debiasing step that removes bias.
 
-## Randomized Response
+Therefore, we define three methods for an interface `DpMechanism`:
 
-> TODO: Specify any primitives required for randomized response mechanisms
+~~~
+class DpMechanism:
+    DataType
+    DebiasedDataType
 
-# DP Mechanisms for VDAFs {#mechanisms}
+    def add_noise(self, data: DataType) -> DataType:
+        """Add noise to a piece of input data. """
+        pass
 
-The section defines a generic interface for DP mechanisms VDAFs.
+    def sample_noise(self, num_reps: int, dimension: int) -> DataType:
+        """
+        Sample noise for `num_reps` number of times,
+        with `dimension`.
+        """
+        pass
 
-> TODO: Junye to spell out the interface of dp mechanisms, "Noise Mechanisms to Accomplish DP"
+    def debias(self,
+               agg: DataType,
+               meas_count: int) -> DebiasedDataType:
+        """
+        Debias the data due to the added noise.
+        This doesn't apply to all noises. Some Client-DP mechanisms
+        need this functionality.
+        """
+        pass
+~~~
 
-## Client-DP for Prio3Histogram
+## Discrete Laplace
+
+> TODO: Specify a Laplace sampler
 
-> TODO: Concretely describe a Client-DP mechanism for Prio3Histogram.
+## Discrete Gaussian {#discrete-gaussian}
 
-## Aggregator-DP for Prio3Histogram
+> TODO: Specify a Gaussian sampler
 
-> TODO: Concretely describe an Aggregator-DP mechanism for Prio3Histogram.
+## Randomized Response {#rr}
+
+> TODO: Specify a randomized response mechanism
+
+# DP Policies for VDAFs {#policies}
+
+The section defines a generic interface for DP policies on VDAFs.
+
+We will define an interface `DpPolicy` that composes the following:
+
+* An optional Client-DP mechanism that adds noise to Clients' measurements.
+
+* An optional Aggregator-DP mechanism that adds noise to an Aggregator's
+  aggregate share, based on the number of measurements, and the minimum
+  batch size.
+
+* An optional debiasing step that removes the bias in DP mechanisms (i.e.
+  `DpMechanism.debias`).
+
+The composition of Client- and Aggregator-DP mechanisms defines the DP
+policy for a VDAF, and enforces the DP guarantee.
+
+~~~
+class DpPolicy:
+    Measurement
+    AggregateShare
+    AggregateResult
+    DebiasedAggregateResult
+
+    def add_client_dp(self,
+                      meas: Measurement,
+                      ) -> Measurement:
+        """
+        Add noise to measurement, if required by the Client-DP
+        mechanism.
+        """
+        pass
+
+    def add_aggregator_dp(self,
+                          agg_share: AggregateShare,
+                          meas_count: Unsigned,
+                          min_batch_size: Unsigned,
+                          ) -> AggregateShare:
+        """
+        Add noise to aggregate share, if required by the Aggregator-DP
+        mechanism.
+        """
+        pass
+
+    def debias_aggregate_result(self,
```

Comment:
@cjpatton Is it worth being explicit here (`debias_aggregate_result` vs `debias`), if the argument takes in an aggregate result?

Reply:
Good catch, yes debias_aggregate_result is better IMO. Fixing the slides.